app/models/container_project.rb

```diff
@@ -10,22 +10,22 @@ class ContainerProject < ApplicationRecord
   include Purging
   belongs_to :ext_management_system, :class_name => "ManageIQ::Providers::ContainerManager", :foreign_key => "ems_id", :inverse_of => :container_projects
   has_many :container_groups, -> { active }, :inverse_of => :container_project
-  has_many :container_routes
-  has_many :container_replicators
-  has_many :container_services
+  has_many :container_routes, :dependent => :destroy
```

Comment:
Many 10s of these.  Maybe handled by event handling or just lower lifecycle churn.

Reply:
`container_routes` are a top-level managed entity that should be deleted by refresh when they are removed from k8s